pkg/storaged/devices.jsx

```diff
@@ -37,14 +37,16 @@ const _ = cockpit.gettext;
 class StoragePage extends React.Component {
     constructor() {
         super();
-        this.state = { path: cockpit.location.path };
+        this.state = { inited: false, slow_init: false, path: cockpit.location.path };
         this.on_client_changed = () => { this.setState({}) };
         this.on_navigate = () => { this.setState({ path: cockpit.location.path }) };
     }
 
     componentDidMount() {
         this.props.client.addEventListener("changed", this.on_client_changed);
         cockpit.addEventListener("locationchanged", this.on_navigate);
+        client.init(() => { this.setState({ inited: true }) });
+        window.setTimeout(() => { this.setState({ slow_init: true }) }, 1000);
```

Comment:
I think this should remember the timer and `client.init()`'s promise handler should stop it. Or, if you prefer, only setState if inited is still false.

Otherwise you'll set slow_init after the page already initialized. It's not a major issue, but does cause an extra page render.

Reply:
> Or, if you prefer, only setState if inited is still false.

Yes, that seems easy and straightforward.  Done.